projects/components/src/table/table.component.ts

```diff
@@ -648,6 +657,53 @@ export class TableComponent
     this.changeDetector.detectChanges();
   }
 
+  private downloadCsv(): void {
+    combineLatest([this.columnConfigs$, this.filters$])
+      .pipe(
+        switchMap(([columnConfigs, filters]) =>
+          (!isNil(this.data)
+            ? this.data.getData({
+                columns: columnConfigs,
+                position: { limit: 1000, startIndex: 0 },
+                filters: filters ?? []
+              })
+            : of({ data: [], totalCount: 0 })
+          ).pipe(
+            take(1),
+            map((response: TableDataResponse<TableRow>) => response.data),
+            map(rows => {
+              const csvGeneratorMap = new Map(
+                columnConfigs
+                  .filter(
+                    column => !isNil(column.csvGenerator) && !(column.csvGenerator instanceof TableCellSkipCsvGenerator)
```

Comment:
two ways of modeling the same thing, so could just skip the skip generator and omit only.

Reply:
fixed